olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionedLayerClient.h

```diff
@@ -34,20 +34,19 @@ namespace olp {
 namespace dataservice {
 namespace read {
 
-using DataResult = model::Data;
-using DataResponse = client::ApiResponse<DataResult, client::ApiError>;
-using DataResponseCallback = std::function<void(DataResponse response)>;
-
 /**
- * @brief The VersionedLayerClient aimed to acquire data from OLP services.
+ * @brief The VersionedLayerClient is responsible for reading data from an OLP
+ * versioned layer.
  */
 class DATASERVICE_READ_API VersionedLayerClient final {
  public:
   /**
    * @brief Sets up \c OlpClientSettings and \c HRN to be used during requests.
+   * @param client_settings to be used during outgoing requests.
+   * @param catalog as \c HRN identifier to be queried by requests.
    */
   VersionedLayerClient(olp::client::OlpClientSettings client_settings,
```

Comment:
The settings are more important to the class since without them nothing will be operating at all (networking)
layer_id will be moved to constructor once it will disappear from DataRequest.

Reply:
This is the current limitation but we do not assume this will be the case in the future. And if the user is ok with the default value then he can leave this parameter out.
We want to lock the API by 0.8.0. After that no API breaks without obsoletion period, downwards compatibility, branching, etc.
We know we need the layer now and it is much easier to deprecate the DataRequest field and remove it in the next, once we removed GetData from CatalogClient. Please add.

Reply:
This will confuse user, which layer is to be used.
Seems like we need to finalize DataRequest before 0.8 then.